src/main/java/com/google/roomies/data/Listing.java

```diff
@@ -0,0 +1,201 @@
+  /**
+  * Describes the two different lease types for a listing.
+  * If input is not one of these two options, servlet throws an error.
+  */
+  enum LeaseType {
+    YEAR_LONG, 
+    MONTH_TO_MONTH;
+  }
+  abstract Optional<String> documentId();
+  abstract Optional<Timestamp> timestamp();
+  abstract String title();
+  abstract String description();
+  abstract Date startDate();
+  abstract Date endDate();
+  abstract LeaseType leaseType();
+  abstract int numRooms();
+  abstract int numBathrooms();
+  abstract int numShared();
+  abstract int numSingles();
+  abstract Money sharedPrice();
+  abstract Money singlePrice();
+  abstract Money listingPrice();
+
+  public static Builder builder() {
+    return new AutoValue_Listing.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder implements Serializable {
+    abstract Builder setTimestamp(Optional<Timestamp> timestamp);
+    abstract Builder setDocumentId(Optional<String> documentId);
+    abstract Builder setTitle(String title);
+    abstract Builder setDescription(String description);
+    abstract Builder setStartDate(Date startDate);
+    abstract Builder setEndDate(Date endDate);
+    abstract Builder setLeaseType(LeaseType leaseType);
+    abstract Builder setNumRooms(int numRooms);
+    abstract Builder setNumBathrooms(int numBathrooms);
+    abstract Builder setNumShared(int numShared);
+    abstract Builder setNumSingles(int numSingles);
+    abstract Builder setSharedPrice(Money sharedPrice);
+    abstract Builder setSinglePrice(Money singlePrice);
+    abstract Builder setListingPrice(Money listingPrice);
+
+    abstract LeaseType leaseType();
+    abstract int numRooms();
+
+    abstract Listing build();
+    
+    /**
+    * Throws IllegalArgumentException if input does not match a LeaseType enum 
+    * value (case sensistive).
+    */
+    public Builder setLeaseType(String leaseType) {
+      setLeaseType(LeaseType.valueOf(leaseType));
+      return this;
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumRooms(String numRooms) {
+      setNumRooms(Integer.parseInt(numRooms));
+      return this;
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumBathrooms(String numBathrooms) {
+      setNumBathrooms(Integer.parseInt(numBathrooms));
+      return this;     
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumShared(String numShared) {
+      setNumShared(Integer.parseInt(numShared));
+      return this;     
+    }
+
+    /**
+    * Throws NumberFormatException if input is not parseable.
+    */
+    public Builder setNumSingles(String numSingles) {
+      setNumSingles(Integer.parseInt(numSingles));
+      return this;     
+    }
+
+    public Builder setStartDate(String startDate) throws ParseException {
```

Comment:
This is expecting the string to be in some particular date format, should call out explicitly what it is.

Same for all these builder methods which accept strings.